src/DataStructures/Variables.hpp

```diff
@@ -841,8 +844,21 @@ struct NumberOfPoints<Variables<TagList>> {
 }  // namespace MakeWithValueImpls
 
 namespace db {
+// Enable subitems for ::Tags::Variables and derived tags (e.g. compute tags).
+// Other tags that hold a `Variables` don't expose the constituent tensors as
+// subitems by default.
+namespace Variables_detail {
+template <typename TagsList>
+constexpr std::true_type is_a_variables_tag(::Tags::Variables<TagsList>&&) {
+  return {};
+}
+constexpr std::false_type is_a_variables_tag(...) { return {}; }
+template <typename Tag>
+static constexpr bool is_a_variables_tag_v =
+    decltype(is_a_variables_tag(std::declval<Tag>()))::value;
+}  // namespace Variables_detail
 template <typename Tag>
-struct Subitems<Tag, Requires<tt::is_a_v<Variables, typename Tag::type>>> {
+struct Subitems<Tag, Requires<Variables_detail::is_a_variables_tag_v<Tag>>> {
```

Comment:
why do you need the above; isn't this just `tt:is_a_v<::Tags::Variables, Tag>` ?

Reply:
No, this includes both the `Tags::Variables` template _and_ classes that inherit from the `Tags::Variables` template. It's needed for compute items inheriting from `Tags::Variables`.

Reply:
Since both @kidder  and I were confused, maybe add a comment why this isn't using `is_a_v` or `is_base_of`?

Reply:
sure, done